Enable Stargate By Default

jdonenine opened the pull request:
<!--  Thanks for sending a pull request!  Before submitting:

1. Read our CONTRIBUTING.md guide
2. Rebase your PR if it gets out of sync with main
-->

**What this PR does**:

* Enable Stargate by default.
* Enable Stargate Cassandra ingress by default
* Enable Stargate GraphQL Playground ingress by default
* Disable direct Cassandra ingress by default

Docs change removes the blurb in the getting-started section specific to Stargate and instead adds a section generally discussing our approach to the default values.  This is a very simple approach for the moment that I expect we'll expand on as we get into some of the meaty doc tasks for GA.

**Which issue(s) this PR fixes**:
Fixes #266

**Checklist**
- [x] Changes manually tested
- [x] Chart versions updated (if necessary)
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

bradfordcp reviewed (commented):
These changes look good.

jakerobb commented on charts/k8ssandra/Chart.yaml:
```diff
@@ -3,8 +3,8 @@ name: k8ssandra
-version: 0.46.0
-appVersion: 3.11.7
+version: 0.47.0
+appVersion: 3.11.10
```
What is the significance of changing this version? Guessing: it's just metadata, and that its intent is to say which version of C* we run by default?

jdonenine replied:
Yeah, sorry, this is unrelated, I just noticed that when we switched the default to .10 we didn't update this here.

jakerobb reviewed (approved):
Looks good to me. Just one question that should not block merging.

burmanm reviewed (changes requested):
There are also tests that assume Stargate was disabled by default. They need to be verified and modified to check that it actually is disabled for those cases. Previously tests assumed that if Reaper is disabled - all the extras are. 

For example, the test:

```go
It("enabling auth and medusa with user-defined secret", func() {
```

Checks that "users" part exists, but it will - even if medusa's part fails, since now Stargate adds it. When the test was created, it would have failed, since it disabled reaper.

All tests that assumed something Stargate enablement could modify have to be modified and new tests added if such case with the enabled changes the template rendering.

jdonenine commented:
Thanks for the heads up on those auth related tests, I had fixed the one that was broken because it was testing all of the accounts created, but these definitely not.  I updated all of the tests that were checking the `cassdcv1beta1.CassandraUser` just for existence, in those cases I disabled stargate in those tests, so they continue to test exactly what they were before.

jakerobb commented:
I resolved the version merge conflict for you. :)

jdonenine commented on tests/unit/template_cassdc_test.go:
```diff
@@ -338,8 +343,8 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
-			// No users should exist
-			Expect(cassdc.Spec.Users).To(BeNil())
+			// With Stargate enabled by default, only the stargate user should exist
+			Expect(cassdc.Spec.Users).To(ConsistOf(cassdcv1beta1.CassandraUser{Superuser: true, SecretName: clusterName + "-stargate"}))
```
@burmanm I updated the test you modified in #357 as discussed over there.

burmanm replied:
Ah, the test was designed to verify that empty users never exists (nil, not size 0), since that fails the CassandraDatacenter deployment. Other tests can verify that the correct user appears.

jdonenine replied:
Ah, OK, so then we actually want this one to disable stargate as well.

jdonenine replied:
OK, swapped it to disable stargate for the test and switched back the check to be the nil check.
